### DISCUSSION TITLE
Replacing SVG in top bar

### DISCUSSION THREAD

**Discussion by houstonhaynes** [mui/toolpad#4118] [Help and support]
I couldn't find a "hook" to replace the SVG that's in the top bar - and didn't know if it was me or if that's something intentionally baked into the boilerplate in a way to prevent it being replaced. Thoughts? Thanks! 
![image](https://github.com/user-attachments/assets/666b1469-8280-43a8-85b7-b0800cc87d28)

**Comment by Janpot** [comment]
The `DashboardLayout` inherits the logo from the `AppProvider` branding as per https://mui.com/toolpad/core/react-dashboard-layout/#branding
But we could probably provide overrides at the `DashboardLayout` level. @apedroferreira what do you think?

**Reply by apedroferreira** [reply]
That sounds good and should hopefully be easy to do!

**Reply by houstonhaynes** [reply]
That's great. 

Overall I found the image handling (or lack thereof) in Toolpad to be clunky. When I added @mui-image the navigation icon disappeared (although icons in the nav menu remained) which meant I could no longer close/open the sidebar menu any longer. Very odd.

**Reply by Janpot** [reply]
Would you be able to share a small repo with what you're trying to do?

@apedroferreira Perhaps it makes sense to add an example in the docs with a non-svg icon? And how is the DX when users provide unexpected sizes?

**Reply by apedroferreira** [reply]
> Would you be able to share a small repo with what you're trying to do?

Yes, more information about what exactly what you were trying to do would be helpful! 
What do you mean by adding @mui-image?

> @apedroferreira Perhaps it makes sense to add an example in the docs with a non-svg icon? And how is the DX when users provide unexpected sizes?

I remember testing this with PNG images and it's supposed to adjust their dimensions (mainly width) to fit, I'll recheck it.
We can also add some tests for this kind of thing I suppose (maybe with snapshots too).

**Reply by houstonhaynes** [reply]
> Would you be able to share a small repo with what you're trying to do?
> 

I would but - and this weird - but only in these repos (and I have a handful of front end sites with node_modules in them *without* this issue) the node_modules exclusion in .gitignore get passed over - so all node_modules folders are added, which is a deal-breaker. I've tried fixing it and frankly I'm not sure why but git is just not picking it up.

UPDATE: I opened the repo in VSCode and was prompted *twice* to add node_modules/ to the .gitignore. Then when I saw that folder "go gray" in the editor I closed out and did the honors. VERY strange...

https://github.com/speakez-llc/toolpad2

This was an experiment to see if I could get a mix of public and authenticated pages in the site - but the top bar issue is there too.

**Reply by Janpot** [reply]
If you already committed the node_modules folder before editing to `.gitignore` you may have to remove them again with `git rm -r node_modules`

**Reply by houstonhaynes** [reply]
I appreciate the advice - it was interesting because I did *exactly* that and they were simply re-added - until I went through the bizarre process with code that eventually masked the node_modules folder in the File Explorer view in the app. Honestly I've never seen anything like it - all good now!
